### PR TITLE
Ab#1374 es6 support

### PR DESCRIPTION
**Pull request by PeculiarGoat**
- Added editorconfig.
- Added rollup + buble.

Does this look okay?  Seems to work alright for me.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,14 +44,21 @@
     "deepmerge": "^4.2.1",
     "del": "^5.1.0",
     "font-awesome": "^4.7.0",
+    "headroom.js": "^0.11.0",
```
Comment (additiverich) [inline package.json:47]: Is this actually being used or has it been copypasta'd from someplace else. I know it's in Curzon but

File: package.json [hunk 2]
```diff
@@ -43,14 +44,21 @@
...
     "node-sass": "^4.14.1",
     "npm-run-all": "^4.1.5",
     "postcss-cli": "^7.1.1",
+    "rollup": "^2.23.0",
+    "rollup-plugin-terser": "^6.1.0",
     "s72-kibble": "^0.10.24",
     "watch": "^1.0.2",
     "yargs": "^15.3.1"
+  },
+  "devDependencies": {
```
Comment (additiverich) [inline package.json:60]: From memory you won't need these if Headroom isn't there. They help resolve headroom.min.js in node_modules to an importable ES6 thing.

File: package.json (hunk 1 above)
Comment (additiverich) [inline package.json:47]: YeahNah.gif

File: rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  plugins: [
+    buble({ jsx: 's72.ui.h' }),
+    (production && terser()),
+    commonjs({
+      namedExports: {
+        'node_modules/headroom.js/dist/headroom.min.js': ['Headroom']
+      }
+    }),
+    resolve()
```
Comment (additiverich) [inline rollup.config.js:30]: Almost definitely don't need either of these if you're not relying on external node modules

File: rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import buble from '@rollup/plugin-buble';
+import { terser } from "rollup-plugin-terser";
+import resolve from '@rollup/plugin-node-resolve';
+import commonjs from 'rollup-plugin-commonjs';
```
Comment (additiverich) [inline rollup.config.js:4]: You'll need to tidy these up

**Review by additiverich — CHANGES_REQUESTED** [review]
You don't need Headroom and by inference a couple of other rollup libs

**Review by additiverich — COMMENTED** [review]
At this point if it works for you than you have my blessing and I hope you'll be very happy together

File: package.json [hunk 5]
```diff
@@ -13,10 +13,10 @@
     "css": "node-sass ./output/site/styles/main.scss ./output/site/static/styles/main.css",
     "css:watch": "npm run css -- --watch",
     "css:build": "cd output && postcss ./site/static/styles/main.css --use autoprefixer --use cssnano -r",
-    "js": "concat -o ./output/site/static/scripts/main.js ./output/site/static/js/*.js",
-    "js:watch": "watch 'npm run js' ./output/site/static/js --wait=3",
+    "js": "concat -o ./output/site/static/js/rollup-target.js ./output/site/static/js/local.js ./output/site/static/js/main.js ./output/site/static/js/modernizr-custom.js; rollup -c",
```
Comment (gscragg) [inline package.json:16]: If you import the required files into the main target then you dont need to use concat.

Comment (gscragg) [reply package.json:16]: I think I understand what you are doing here and I don't like it. Perhaps the rollup-target js file should exsts somewhere else?

**Review by gscragg — CHANGES_REQUESTED** [review]
The package scripts need some work.

File: package.json (hunk 5 above)
Comment (PeculiarGoat) [reply package.json:16]: I guess I could make a `temp` directory and output `rollup-target.js` in there (and add it to gitignore).  That will eliminate the need for the watch filter.  And the concat line can just revert to how it was before... concat is still required to combine `local.js` with the `main.js` and `modernizr-custom.js` files from core.

Sound okay?

Comment (gscragg) [reply package.json:16]: Using import statements from local files to core files should remove this requirement.

**Review by gscragg — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (PeculiarGoat) [reply package.json:16]: Ah got it!

**Review by gscragg — APPROVED** [review]

**Pull request merged**